hooks/pre_validations/31-mongo_storage_engine.rb

```diff
@@ -0,0 +1,54 @@
+MONGO_ENGINE_COMPLETE = '/etc/foreman-installer/.mongo_engine_upgrade'.freeze
+MONGO_DIR = '/var/lib/mongodb/'
+
+# Pull MongoDB connection info from answer file
+if answers['katello']
+  MONGO_HOST = param('katello', 'pulp_db_seeds')
+else
+  MONGO_HOST = param('foreman_proxy_content', 'pulp_db_seeds')
+end
+
+def fail_and_exit(message)
+  Kafo::Helpers.log_and_say :error, message
+  kafo.class.exit 1
+end
+
+def disk_space
+  # Check diskspace in /tmp
+  logger.info 'Checking available diskspace in /tmp for upgrade'
+  total_space = `df -k --exclude-type=tmpfs /tmp`.split("\n")[-1].split[3].to_i
+  mongo_size = File.directory?(MONGO_DIR) ? `du -s  /var/lib/mongodb/`.split[0].to_i : 0
```

Comment:
`du -s  /var/lib/mongodb/` should use `MONGO_DIR`

Reply:
Disk space is done in the pre_validation hook, so it is not created yet. To compromise I switch all the folders to /var/tmp so we should be good there.